packages/babel-helper-define-polyfill-provider/src/index.ts

```diff
@@ -41,13 +39,16 @@ function resolveOptions<Options>(
   options: PluginOptions,
   babelApi,
 ): {
-  method: MethodString,
-  methodName: "usageGlobal" | "entryGlobal" | "usagePure",
-  targets: Targets,
-  debug: boolean,
-  shouldInjectPolyfill: ?(name: string, shouldInject: boolean) => boolean,
-  providerOptions: ProviderOptions<Options>,
-  absoluteImports: string | boolean,
+  method: MethodString;
+  methodName: "usageGlobal" | "entryGlobal" | "usagePure";
+  targets: Targets;
+  debug: boolean;
+  shouldInjectPolyfill:
+    | ((name: string, shouldInject: boolean) => boolean)
+    | undefined
+    | null;
```

Comment:
there are some [functional differences between](https://github.com/niieani/typescript-vs-flowtype#maybe--nullable-type) Flow's Maybe type and TS's optional type, so, per my commit message, you'll see similar changes in several places

Reply:
While `?` means "or null or undefined", I believe in all the occurrences in this repo `| undefined` should be enough (unless TS complains obviously)

Reply:
Gotcha. That might take me some time to change all the occurrences to TS's optional type `?` and remove the `| undefined | null`

Reply:
Completed by Nicolò in https://github.com/babel/babel-polyfills/commit/80605cc0084c4adfd6b50c2d626ce2bd169a4236 . See follow-up in https://github.com/babel/babel-polyfills/pull/121#discussion_r845379333